### PR TITLE
chore: add unit tests for PlatformNavigator

### PR DESCRIPTION
**Pull request by katiegoines**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    render(component);
+
+    const platformButton = await screen.getByRole('button');
+    expect(platformButton.getAttribute('aria-expanded')).toBe('false');
+
+    userEvent.click(platformButton);
+    expect(platformButton.getAttribute('aria-expanded')).toBe('true');
+  });
+
+  it('should change platform on selection', async () => {
```
Comment (timngyn) [inline src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:52]: In this test we're not actually changing the platform right? We're just tabbing through the menu to make sure the correct platform is focused?

Maybe we could have a more descriptive test name like
```suggestion
  it('should tab through platforms in dropdown', async () => {
```

**Review by timngyn — APPROVED** [review]
LGTM, just had a small comment to make the test more descriptive, but it's not blocking

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx (hunk 1 above)
Comment (reesscot) [reply src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:52]: +1

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    render(component);
+
+    const platformButton = await screen.getByRole('button');
+    const popover = await screen.getByRole('navigation');
+    const popoverFirstItem = popover.children[0].children[1];
+    userEvent.click(platformButton);
+    userEvent.tab();
+    userEvent.tab();
```
Comment (reesscot) [inline src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:60]: Looks like you need a tests that the tab is focusing on the right element? Below it looks like you're using the popoverFirstItem, which doesn't require tabbing.

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    expect(popoverFirstItem.textContent).toBe('JavaScript');
+    expect(popoverFirstItem.children[0].getAttribute('href')).toBe(
+      '/javascript'
+    );
+    expect(popoverFirstItem.children[0]).toHaveFocus();
```
Comment (reesscot) [inline src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:65]: Can you move this focus selector right below `userEvent.tab` for readability?

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  );
+
+  it('should render the PlatformNavigator component', async () => {
+    render(component);
+
+    const navigator = await screen.findByText('Choose your framework/language');
+    expect(navigator).toBeInTheDocument();
+  });
+
+  it('should show the selected platform as React', async () => {
```
Comment (reesscot) [inline src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:29]: I think you mean default platform?
```suggestion
  it('should show the default platform as React', async () => {
```

File: src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    render(component);
+
+    const navigatorComponent = await screen.findByText(
+      'Choose your framework/language'
+    );
+    const platform =
+      navigatorComponent.parentElement?.getElementsByTagName('button')[0]
```
Comment (reesscot) [inline src/components/PlatformNavigator/__tests__/PlatformNavigator.test.tsx:36]: Could this be simplified to:
```
await screen.findByText(
      'Choose your framework/language'
    ).findByRole('button').textcontent
```
?

**Review by reesscot — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**